horreum-integration/src/test/java/io/hyperfoil/tools/HorreumTestBase.java

```diff
@@ -42,7 +46,10 @@
 
 import io.hyperfoil.tools.horreum.entity.json.Test;
 
-public class HorreumTestBase {
+import io.restassured.RestAssured;
+import io.restassured.specification.RequestSpecification;
+
+public class HorreumTestBase implements BeforeAllCallback, AfterAllCallback, ExtensionContext.Store.CloseableResource  {
```

Comment:
Looks like you made this an extension, it is not used as 'test base' anymore - the name should reflect that.

And about `HorreumTestClientBase` - the original `HorreumTestBase` class is used for Jenkins plugins tests as well, and therefore it should create the infrastructure & start Horreum but **not** initialize anything specific to the resteasy client. Please layer the classes for reuse in the Jenkins plugin tests. And make sure that any module that includes the base class is publishing test-jars.

Reply:
> Looks like you made this an extension, it is not used as 'test base' anymore - the name should reflect that.
 Ok the name will be updated.

> And about HorreumTestClientBase - the original HorreumTestBase class is used for Jenkins plugins tests as well, and therefore it should create the infrastructure & start Horreum but not initialize anything specific to the resteasy client. 
I have had no end of strife (InitialisationError errors) initializing the Resteasy client in multiple places. Doing it instead once and sharing the built instance works.

> Please layer the classes for reuse in the Jenkins plugin tests. 

 I have created a type hierarchy here so that a subsequent integration test (to be added in another commit) avoids creating the HorreumClient.

> And make sure that any module that includes the base class is publishing test-jars.
ok I will add that.